src/v/cloud_storage/remote_segment.cc

```diff
@@ -962,7 +976,9 @@ remote_segment_batch_reader::read_some(
             _parser = co_await init_parser();
         }
 
-        if (ot_state.add_absolute_delta(_cur_rp_offset, _cur_delta)) {
+        if (
+          ot_state.empty()
+          && ot_state.add_absolute_delta(_cur_rp_offset, _cur_delta)) {
```

Comment:
Maybe a comment to explain why this is only valid if the ot_state is empty?  Is the idea that if ot_state is non-empty that it should have already be up-to-date from reading earlier regions in the segment?

It would be good to detect and log an error in the case where `!ot_state.empty()` and `add_absolute_delta` would have hit a `return true` path.  Maybe could pass a flag into add_absolute_delta to have it log an error instead of making a change in those paths.

Reply:
> Is the idea that if ot_state is non-empty that it should have already be up-to-date from reading earlier regions in the segment?

Yes. `add_absolute_delta` is how we initialise the segment on the first read from the segment. After the first read,
the offset translator should be kept up to date by the `add_gap` calls made by `remote_segment_batch_consumer`.

> It would be good to detect and log an error in the case where !ot_state.empty() and add_absolute_delta would have hit a return true path.
I think we log this `remote_segment_batch_consumer::skip_batch_start` now.

Reply:
This is a bit suspicious to me because `ot_state` is initialized once for the read request and can span multiple segments. So the purpose of this call is to "re-sync" offset translation state at the beginning of a segment (and `ot_state` is not necessary empty at this point). This can come in handy if we for some reason have a gap in uploaded data.

OTOH in case of overlapping segments we can just modify `add_absolute_delta` to expect that it can be called with already known offsets. In this case it can simply check that the passed delta is consistent with what is already known by the offset translator.

Reply:
This makes sense to me. I can try it in the followup PR.